cmd/channel/email/main.go

```diff
@@ -38,17 +46,50 @@ func (ch *Email) SendNotification(req *plugin.NotificationRequest) error {
 	}
 
 	var msg bytes.Buffer
-	_, _ = fmt.Fprintf(&msg, "To: %s\n", strings.Join(to, ","))
-	_, _ = fmt.Fprintf(&msg, "Subject: [#%d] %s %s is %s\n\n", req.Incident.Id, req.Event.Type, req.Object.Name, req.Incident.Severity)
-
 	plugin.FormatMessage(&msg, req)
 
-	err := smtp.SendMail(ch.GetServer(), nil, ch.From, to, bytes.ReplaceAll(msg.Bytes(), []byte("\n"), []byte("\r\n")))
+	return enmime.Builder().
+		ToAddrs(to).
+		From(ch.SenderName, ch.SenderMail).
+		Subject(fmt.Sprintf("[#%d] %s %s is %s", req.Incident.Id, req.Event.Type, req.Object.Name, req.Incident.Severity)).
+		Header("Message-Id", fmt.Sprintf("<%d-%x-%s>", req.Incident.Id, time.Now().UnixNano(), ch.SenderMail)).
```

Comment:
I'd just go for a UUID instead of `req.Incident.Id` and `time.Now().UnixNano()` here.

Reply:
You've pushed an outdated version which does not include the changes that were already done in https://github.com/Icinga/icinga-notifications/pull/123#event-11168199610.

(Hint: this is a reply to https://github.com/Icinga/icinga-notifications/pull/123/files#r1414033413, GitHub sometimes doesn't show it clearly if you reply to existing threads in a review.)